packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1318,6 +1323,22 @@ export class TransactionController extends BaseControllerV1<
   startTrackingByNetworkClientId(networkClientId: NetworkClientId) {
     const networkClient = this.getNetworkClientById(networkClientId);
     const { chainId } = networkClient.configuration;
+    if (!chainId) {
+      throw new Error('NetworkClient must have a chainId');
```

Comment:
i would have assumed typing already guarantees this

Reply:
Yeah true. This isn't needed.

Reply:
done here: https://github.com/MetaMask/core/pull/3822/commits/00f5192cff537500fa217bd755e90f733746d878